File: Stack.java
class Stack{

    public static void main(String[] args) {
        int iRet=0;


        StackX Aobj=new  StackX();
        Aobj.Push(11);
        Aobj.Push(21);
        Aobj.Push(51);
        Aobj.Push(101);
        Aobj.Push(111);
        Aobj.Push(121);
        
        

        Aobj.Display();
        iRet=Aobj.Count();
        System.out.println("number of elements in stack are : "+iRet);

        Aobj.Pop();
        Aobj.Display();
        iRet=Aobj.Count();
        System.out.println("number of elements in stack are : "+iRet);

        

        
    }
}
class node{

   public int data;
  public   node next;

   public node(int iValue){

        data=iValue;
        next=null;
    }

}

class StackX{
     
    public node First;
    public int iCount;

    public StackX(){

        System.out.println("inside constructor: ");

        First=null;
        iCount=0;

    }
    public void Push(int iNo){

        node newn=null;
        newn=new node(iNo);

        if(First==null){

            First=newn;

        }
        else{

            newn.next=First;
            First=newn;

        }
        iCount++;
    }

    public void Pop(){

        if(First==null){
            System.out.println("LL is Already Empty: ");
            return;
        }
        else if(First.next==null){
            First=null;

        }
        else{

            First=First.next;
        }
        iCount--;
    }

    public void Display(){

        node temp=First;

       while (temp !=null) { 
        System.out.print("|"+temp.data+"|->");
        temp=temp.next;
       }
       System.out.print("| null |\n");

      
       
    }
    public int Count()
    {
       return iCount;
   }
}

File: Stack.c
#include<stdio.h>
#include<stdlib.h>


struct node{
    int data;
    struct node*next;

};
typedef struct node NODE;
typedef struct node* PNODE;
typedef struct node** PPNODE;


void Push(PPNODE First,int No){
 
   PNODE newn=NULL;

   newn=(PNODE)malloc(sizeof(NODE));
    newn->data=No;
    newn->next=NULL;

    if(*First==NULL){

        *First=newn;

    }
    else{

        newn->next=*First;
        *First=newn;

    }

     
}
int Pop(PPNODE First){

  int iValue=0;
  PNODE temp=NULL;

  if(*First==NULL){

    printf("unable top pop the element as stack is empty : ");

    return -1;

  }
  else{

    temp=*First;
    
    iValue=(*First)->data;
    *First=(*First)->next;
    free(temp);
  }
return iValue;



}
void Display(PNODE First){
     printf("Elements of stack are : \n");
 

    while(First!= NULL)
    {
        printf("%d\n",First->data);
        First= First -> next;
    }
    
    printf("\n");
}
int Count(PNODE First){
int iCnt=0;

    while(First!= NULL)
    {
       iCnt++;

        First= First -> next;
    }
    
return iCnt;

}
int main(){

PNODE Head=NULL;
int iRet=0;
int iRetX=0;

Push(&Head,51);
Push(&Head,21);
Push(&Head,11);
Push(&Head,101);

Display(Head);
iRet=Count(Head);
printf("Number of elements of stack  %d\n",iRet);


iRetX=Pop(&Head);
printf("Poped element of stack is : %d\n",iRetX);
Display(Head);
iRet=Count(Head);
printf("Number of elements of stack : %d\n",iRet);

Push(&Head,111);

Display(Head);
iRet=Count(Head);
printf("Number of elements of stack : %d\n",iRet);



    return 0;

}